### DISCUSSION TITLE
How to keep a package's node_modules isolated in a workspace managed by pnpm?

### DISCUSSION THREAD

**Discussion by ericxinzhang** [pnpm/pnpm#5378] [Q&A]
My project folder structure looks as following:

```
|-app1
|    node_modules
|    package.json
|-app2
|    node_modules
|    package.json
|-node_modules
|-package.json
|-pnpm-lock.yaml
|-pnpm-workspace.yaml
```
The workspace package.json does not have any dependencies.
Initially the pnpm-workspace.yaml only has app1. I ran `pnpm i` and `pnpm -r test` and app1's test passed successfully.

I then included app2 into pnpm-workspace.yaml and ran `pnpm i` again. I could see that `pnpm-lock.yaml` was heavily modified. When I ran `npm -r test` again app1's test was throwing error from jest/expect modules. 

I removed app2 from pnpm-workspace.yaml, deleted node_modules under app1 and then ran install and test again, the test passed.

It looks to me that app1's node_modules got polluted when app2's dependencies were installed. I am new to pnpm and thought pnpm should keep app1's dependency intact/isolated in this case. After all nothing within app1 has changed. 

Has anyone had same problem and what is the solution?

**Comment by rhyek** [comment]
I've run into multiple issues in the past related somewhat to this. I would also like a configuration option to have each apps's node_modules and pnpm-lock isolated.

**Reply by rhyek** [reply]
https://pnpm.io/npmrc#shared-workspace-lockfile seems to do the trick
